This short introduction uses [Keras](https://www.tensorflow.org/guide/keras/overview) to:

1. Build a neural network that classifies images.
2. Train this neural network.
3. And, finally, evaluate the accuracy of the model.

In [1]:
import tensorflow as tf

Load and prepare the [MNIST dataset](http://yann.lecun.com/exdb/mnist/). Convert the samples from integers to floating-point numbers:

In [2]:
mnist = tf.keras.datasets.mnist

(x_train, y_train), (x_test, y_test) = mnist.load_data()
x_train, x_test = x_train / 255.0, x_test / 255.0

Build the `tf.keras.Sequential` model by stacking layers. Choose an optimizer and loss function for training:

In [3]:
model = tf.keras.models.Sequential([
  tf.keras.layers.Flatten(input_shape=(28, 28)),
  tf.keras.layers.Dense(128, activation='relu'),
  tf.keras.layers.Dropout(0.2),
  tf.keras.layers.Dense(10, activation='softmax')
])

In [4]:
model.compile(optimizer='adam',
              loss='sparse_categorical_crossentropy',
              metrics=['accuracy'])

The `Model.fit` method adjusts the model parameters to minimize the loss: 

In [5]:
model.fit(x_train, y_train, epochs=5)

Train on 60000 samples
Epoch 1/5
60000/60000 [==============================] - 7s 120us/sample - loss: 0.2916 - accuracy: 0.9151
Epoch 2/5
60000/60000 [==============================] - 6s 107us/sample - loss: 0.1388 - accuracy: 0.9586
Epoch 3/5
60000/60000 [==============================] - 7s 109us/sample - loss: 0.1053 - accuracy: 0.9681
Epoch 4/5
60000/60000 [==============================] - 6s 108us/sample - loss: 0.0887 - accuracy: 0.9731
Epoch 5/5
60000/60000 [==============================] - 6s 107us/sample - loss: 0.0757 - accuracy: 0.9762


In [6]:
model.evaluate(x_test,  y_test, verbose=2)

10000/1 - 1s - loss: 0.0424 - accuracy: 0.9761


[0.08233443364389241, 0.9761]

#### The image classifier is now trained to ~98% accuracy on this dataset